Okta documentation. (#25580)
gravitational/teleport: docs/pages/application-access/okta/guide.mdx

---
title: Setting up the Okta service
description: How to set up the Okta service with some basic configurations.
---

> This guide will help you set up the Okta service and provide some basic

Reviewer: Similarly, _why_ might I want to provide basic configurations to the Okta service? What can I actually accomplish by doing so?

Reviewer 2: @mdwn I agree with Zac here, let's give a few bullet points of what you can actually achieve by setting up Okta integration: import apps/groups, use access requests to provide temporary access, etc.

configurations to enable it.

## Prerequisites

(!docs/pages/includes/edition-prereqs-tabs.mdx!)

- (!docs/pages/includes/tctl.mdx!)

- An Okta organization with an [Okta API token](https://developer.okta.com/docs/guides/create-an-api-token) provisioned.

> - A running Teleport auth/proxy instance.

Reviewer: Should we just say a Teleport cluster?

- An instance where you will run the Okta service. This can live anywhere
  with outbound access to Okta and must be running Linux.

## Step 1/X. Create Okta import rules.

> Before setting up the Okta service, we should create Okta import rules in order

Reviewer 2: Nit: Instead of "we", let's say "you" everywhere.

to ensure that our Okta applications are set up with appropriate labeling before
we attempt to synchronize them. First, we'll need to get the group and application
IDs from Okta. The easiest way to do this is to use the <Var name="okta-api-token" />
provisioned earlier.

To get group IDs:

```

> $ curl -H -H "Authorization: SSWS <Var name="okta-api-token" />" <Var name="okta-endpoint-url" />/api/v1/groups | jq '[.[] |  {"id": .id, "label": .profile.name}]'

Reviewer 2: ```suggestion
$ curl -H "Authorization: SSWS <Var name="okta-api-token" />" <Var name="okta-endpoint-url" />/api/v1/groups | jq '[.[] |  {"id": .id, "label": .profile.name}]'
```

...
[
  {
    "id": "111111",
    "label": "Okta Admin Console"
  },
  {
    "id": "222222",
    "label": "Okta Browser Plugin"
  },
  {
    "id": "333333",
    "label": "Okta Dashboard"
  },
  {
    "id": "444444",
    "label": "SAML test"
  },
  ...
]
```

From here, you can copy the IDs from the output of the `curl` commands and use
these to create the Okta import rule. Put this in a file called `okta-import-rule.yaml`:

```yaml
kind: okta_import_rule
version: v1
metadata:
  name: import-rule
  description: "Okta import rule for all groups and applications"
spec:
  mappings:
  - match:
    - app_ids: ["111111", "222222", "333333"] # Replace these with your Okta application IDs.
    add_labels:
      app_label: app-label-value
  - match:
    - group_ids: ["111111", "222222", "333333"] # Replace these with your Okta group IDs.
    add_labels:
      group_label: group-label-value
```

This will ensure that the applications and groups, once synchronized to Teleport,
will be given the specified labels.

## Step 2/3. Set up the Okta service.

> In this step, we'll set up the Okta service.

Reviewer 2: I'd briefly explain here what Okta service does and why it's needed. Mention that it's the service that holds Okta API key and interacts with Okta API to import app/groups and update memberships, and that it connects back to Teleport cluster over reverse tunnel.

(=presets.tokens.first=)
```

On the host where you will install the Teleport Okta Service, create a
file called `/tmp/token` that consists only of your token:

```code
$ echo <Var name="join-token" /> | sudo tee /tmp/token
```

### Configure the Okta service

First, on the host where you will run the Okta service , create a file at
<Var name="okta-api-token-path" /> with the contents of the API token.

Next, on the same host, create a file at `/etc/teleport.yaml` with the
following content:

```yaml
version: v3
teleport:
  join_params:
    token_name: "/tmp/token"
    method: token
  proxy_server: "teleport.example.com:443"
auth_service:
  enabled: no
proxy_service:
  enabled: no
ssh_service:
  enabled: no
okta_service:
  enabled: yes
  api_endpoint: <Var name="okta-endpoint-url" />
  api_token_path: <Var name="okta-api-token-path" />
```

Edit `/etc/teleport.yaml` to replace `teleport.example.com:443` with the host
and port of your Teleport Proxy Service or Teleport Cloud tenant, e.g.,
`mytenant.teleport.sh:443`.

### Run the Teleport Application Service

(!docs/pages/includes/start-teleport.mdx service="the Teleport Application Service"!)

> ## Step 3/3. Connect to the Teleport web UI

Reviewer 2: Should mention how to request access to Okta app/group as well.

Navigate to the application access page within the Teleport web UI to see the list of
of synchronized Okta applications. If you're unable to see the applications, please refer
to the application [Access Controls](../controls.mdx) and update your roles to match the

> labels given to your applications by the above specified Okta import rules.

Reviewer: I read all the way to the end, and even if I've followed the steps I still don't quite understand what I've done or why I would have wanted to do it to begin with.

I think we need to reframe this guide to be more about the task a user can accomplish with the Okta service.

---
title: Setting up the Okta Service
description: How to set up the Okta Service with some basic configurations.
---

> Through the use of the Okta Service, Teleport is now able display and grant
> access to Okta applications and user groups. Okta applications can now be
> displayed and accessed through Teleport's application access UI, and access
> to these applications along with user groups can be managed by Teleport's
> RBAC along with access requests.

Reviewer 2: ```suggestion
Teleport can import and grant access to Okta applications and user groups. Okta applications
can be accessed through Teleport's application access UI, and access to these applications along
with user groups can be managed by Teleport's RBAC along with access requests.
```
